pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/bloom/OffHeapGuavaBloomFilterReader.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.index.readers.bloom;
+
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Longs;
+import org.apache.pinot.core.segment.index.readers.BloomFilterReader;
+import org.apache.pinot.core.segment.memory.PinotDataBuffer;
+
+
+/**
+ * Off-heap reader for guava bloom filter.
+ * <p>The behavior should be aligned with {@link com.google.common.hash.BloomFilter}.
```

Comment:
So, our current implementation depends on `com.google.common.hash.BloomFilter` for bloom filter creation while we do have a custom offheap reader. This means that we need to adjust our offheap reader implementation if Guava changes the internal implementation details.

I don't think that they will change the underlying convention often; however, we should also consider having our own custom bloom filter creator whose behavior aligns with this reader (basically mimic the existing guava) if the work is not too much. In that case, we won't need any external dependency and don't need to worry about aligning.

Reply:
Good point. While they should not change that though, otherwise all existing bloom filter generated with guava library will stop working.
Added a TODO to have our own implementation and remove the dependency on guava bloom filter.

Reply:
What I meant was that we can generate the same bytes that Guava bloom filter library would generate and use the existing offheap reader which already knows how to read the guava format. Anyway, adding TODO comment should be good enough for now.